enhancements/etcd/disaster-recovery-with-ceo.md

```diff
@@ -19,86 +19,130 @@ see-also:
+In the prior releases, we restored the backup on all masters simultaneously for the cluster to come backup properly. With this release we implement a different approach of bringing a control plane up with a single node etcd cluster and then scaling it up to three.
+
+This new approach involves two steps:
+1. Get the first etcd member running. 
+
+    1. move the existing etcd pod yaml out of kubelet manifest directory and move etcd data directory into a different 
+  location on all the master nodes. This will stop all the etcds leading to total quorum loss and remove all of the data
+   directories, making the backup a single source of truth.
+    1. Select a master node to run the restore operation on (recovery node) and make sure etcd backup file in present
+   on the recovery node
+    1. copy the snapshot file on a temporary location and move the copy to backup directory to 
+  /var/lib/etcd-backup/snapshot.db on the recovery node
+    1. move the kube-apiserver backed up revision back into kubelet manifest directory and delete other revisions and 
+  related resources(certs and configmaps). Because the restoration is happening from a backed up data, the revision of
+  kube-apiserver pod yaml needs to match the revision that the operator was on while taking the backup.
+    1. copy the restore-etcd-pod.yaml from backed up resources to /etc/kubernetes/manifests/etcd-pod.yaml. This will 
+  restore a single member control plane using the backed up snapshot.db and then start the etcd pod bringing the control
+  plane up. 
+
+2. Get the operator to deploy static pods that can decide that they are being restored, so they don't start on the existing data-dir.
+    1. force CEO redeployment via `oc patch forceredpeloymment=reason-time`
+    1. When CEO redeployment is forced, the existing nodes will start with new pods, a scale up similar to bootstrap will take place
+    1. CEO will redeploy the modified pod on recovery master node to the current revision.
+
+### Recovering a single unhealthy master
+
+If the cluster-etcd-operator status shows `Degraded` condition as `true` with the status of a single `unhealthy` member, then the etcd member could be in one of the three scenarios:
+
+1. The node in question is down and expected to come back up. 
+
+    In this case, admin doesn't need to do anything.
+    
+1. The node is up and generally healthy, but etcd is unhealthy for some reason. 
+
+    In this case, go to the "forcing an existing member to rejoin"
+    
+1. The node is down and not expected to come back. 
+
+    In this case, remove the etcd member followed by "adding a new master node"
+    
+### Forcing an existing member to rejoin
+
+If the node is healthy, but somehow etcd is reporting problems, we suggest to stop the pod, remove the member and force redeployment, so that cluster-etcd-operator can regenerate static-pod-resources and restart the pod.  The sequence of commands can be listed as below:
```

Comment:
We need to be careful here and define what "healthy" and "problems" mean. I think the only reason to remove etcd is if it has experienced catastrophic failure  'C |'. So that it is crash looping and resolution is not possible. Member removal should be viewed as the last case scenario. A transient failure in the logs should not be viewed as grounds for adjusting membership. We just need to be very very clear these situations are rare.

Reply:
Updated. Can you take a look at it again?